examples/playwright/src/theia-notebook-editor.ts

```diff
@@ -0,0 +1,164 @@
+
+import { Locator } from '@playwright/test';
+import { join } from 'path';
+import { TheiaApp } from './theia-app';
+import { TheiaEditor } from './theia-editor';
+import { TheiaNotebookCell } from './theia-notebook-cell';
+import { TheiaNotebookToolbar } from './theia-notebook-toolbar';
+import { TheiaQuickCommandPalette } from './theia-quick-command-palette';
+import { TheiaToolbarItem } from './theia-toolbar-item';
+import { OSUtil, normalizeId, urlEncodePath } from './util';
+
+export class TheiaNotebookEditor extends TheiaEditor {
+
+    constructor(filePath: string, app: TheiaApp) {
+        // shell-tab-notebook::file://<path>
+        // notebook:file://<path>
+        super({
+            tabSelector: normalizeId(`#shell-tab-notebook:file://${urlEncodePath(join(app.workspace.escapedPath, OSUtil.fileSeparator, filePath))}`),
+            viewSelector: normalizeId(`#notebook:file://${urlEncodePath(join(app.workspace.escapedPath, OSUtil.fileSeparator, filePath))}`)
+        }, app);
+    }
+
+    protected viewLocator(): Locator {
+        return this.page.locator(this.data.viewSelector);
+    }
+
+    tabLocator(): Locator {
+        return this.page.locator(this.data.viewSelector);
+    }
+
+    override async waitForVisible(): Promise<void> {
+        await super.waitForVisible();
+        // wait for toolbar being rendered as it takes some time to load the kernel data.
+        await this.notebookToolbar().waitForVisible();
+    }
+
+    /**
+     * @returns The main toolbar of the notebook editor.
+     */
+    notebookToolbar(): TheiaNotebookToolbar {
+        return new TheiaNotebookToolbar(this.viewLocator(), this.app);
+    }
+
+    /**
+     * @returns The name of the selected kernel.
+     */
+    async selectedKernel(): Promise<string | undefined | null> {
+        const kernelItem = await this.toolbarItem('notebook.selectKernel');
+        if (!kernelItem) {
+            throw new Error('Select kernel toolbar item not found.');
+        }
+        return this.notebookToolbar().locator.locator('#kernel-text').innerText();
+    }
+
+    /**
+     *  Allows to select a kernel using toolbar item.
+     * @param kernelName  The name of the kernel to select.
+     */
+    async selectKernel(kernelName: string): Promise<void> {
+        // TODO should we use NotebookCommands.SELECT_KERNEL_COMMAND.id? Need a dependency...
+        await this.triggerToolbarItem('notebook.selectKernel');
+        const qInput = new TheiaQuickCommandPalette(this.app);
+        const widget = await this.page.waitForSelector(qInput.selector, { timeout: 5000 });
+        if (widget && !await qInput.isOpen()) {
+            throw new Error('Failed to trigger kernel selection');
+        }
+        await qInput.type(kernelName, true);
+        await qInput.hide();
+    }
+
+    async availableKernels(): Promise<string[]> {
+        // TODO should we use NotebookCommands.SELECT_KERNEL_COMMAND.id? Need a dependency...
```

Comment:
As above, I'd have a slight preference to introduce a namespace with constants in this page object file.